doc/source/dsintro.rst

```diff
@@ -507,9 +507,41 @@ of one argument to be called on the ``DataFrame``. A *copy* of the original
 DataFrame is returned, with the new values inserted.
 
 .. warning::
+  Starting from Python 3.6 ``**kwargs`` is an ordered dictionary and ``assign`` 
+  respects the order of the keyword arguments. It is allowed to write
 
-  Since the function signature of ``assign`` is ``**kwargs``, a dictionary,
-  the order of the new columns in the resulting DataFrame cannot be guaranteed
+  .. ipython::
+      :verbatim:
+
+      In [1]: # Allowed for Python 3.6 and later
+               df.assign(C = lambda x: x['A'] + x['B'],
+                         D = lambda x: x['A'] + x['C'])
+
+  This may subtly change the behavior of your code when you're
+  using ``.assign()`` to update an existing column. Prior to Python 3.6,
+  callables referring to other variables being updated would get the "old" values
+
```

Comment:
This previous does not need to be here, ONLY in the whatsnew new

Reply:
remove from here down

Reply:
Ok, so implicitly, if one uses pandas 0.23 it is assumed that also python  >= 3.6, right?

Reply:
why would you say that?

remove it because it is redundant (you said it above)